### PR TITLE
A4A: Add Hosting section.

### PR DESCRIPTION
**Pull request by jkguidaven**
This pull request adds a hosting section to the Marketplace which includes skeleton pages for both Pressable and WPCOM.

Screenshot 2024-03-21 at 7 16 49 PM
Screenshot 2024-03-21 at 7 16 56 PM
Screenshot 2024-03-21 at 7 17 03 PM

Closes https://github.com/Automattic/automattic-for-agencies-dev/issues/73

## Proposed Changes

* Transform current product listing components and hooks into shared code, including **Filter search** and **Listing section** and the **useProductAndPlans** hook.
* Update **useProductAndPlans** hook to include the **Pressable** filter.
* Implement a new hosting card that displays Hosting provider information in the main Hosting section.* Add **Pressable** and **WPCOM** overview pages' skeleton.

## Testing Instructions
* Use the live link and go to `/marketplace/hosting`
* Confirm that the Hosting section is rendered properly and loads the Pressable and Wordpress.com cards.
* Click the explore buttons and confirm it redirects to the correct Hosting overview pages.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/a8c-for-agencies/sections/marketplace/hosting-overview/hooks/use-hosting-description.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+} {
+	const translate = useTranslate();
+
+	return useMemo( () => {
+		let description = '';
+		let name = '';
+
+		switch ( slug ) {
+			case 'pressable-hosting':
+				name = 'Pressable';
```
Comment (jkguidaven) [inline client/a8c-for-agencies/sections/marketplace/hosting-overview/hooks/use-hosting-description.ts]: I am not sure if the Pressable and Wordpress.com needs to be a translatable string.

Comment (yashwin) [reply client/a8c-for-agencies/sections/marketplace/hosting-overview/hooks/use-hosting-description.ts]: I think it is better to translate.

File: client/a8c-for-agencies/sections/marketplace/hosting-overview/hosting-card/index.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+				className="hosting-card__explore-button"
+				href={ getHostingPageUrl( plan.family_slug ) }
+				onClick={ onExploreClick }
+				primary
+			>
+				{ translate( 'Explore %(hosting)s plans', {
+					args: {
+						hosting: name,
+					},
+				} ) }
```
Comment (yashwin) [inline client/a8c-for-agencies/sections/marketplace/hosting-overview/hosting-card/index.tsx]: I think it is better to leave a comment on what exactly `hosting` is here to make it easy for the translators. Also, as per my above comment, it is better to translate the name as the value of the name is dynamic in this translation method.

File: client/a8c-for-agencies/sections/marketplace/lib/hosting.tsx [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+ * Get the logo for a hosting provider
+ * @param {string} slug - Hosting provider slug
+ * @returns {Element} - Hosting provider logo
+ */
+export function getHostingLogo( slug: string ) {
+	switch ( slug ) {
+		case 'pressable-hosting':
+			return (
+				<svg
+					width="198"
```
Comment (yashwin) [inline client/a8c-for-agencies/sections/marketplace/lib/hosting.tsx]: I think we can keep these SVG files here: [client/assets/images](https://github.com/Automattic/wp-calypso/blob/e8d66114b31392a90f4fb08d1601d98c3b456c95/client/assets/images). What do you think?

File: client/a8c-for-agencies/sections/marketplace/lib/hosting.tsx [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+	}
+
+	return A4A_MARKETPLACE_HOSTING_LINK;
+}
+
+/**
+ * Get the logo for a hosting provider
+ * @param {string} slug - Hosting provider slug
+ * @returns {Element} - Hosting provider logo
+ */
```
Comment (yashwin) [inline client/a8c-for-agencies/sections/marketplace/lib/hosting.tsx:44]: [Nitpick] 

Screenshot 2024-03-26 at 2 03 48 PM

**Review by yashwin — COMMENTED** [review]
Good on this, @jkguidaven!

Overall looks good and works as described. I have left a few comments. Let me know what you think.

**Comment by jkguidaven** [comment]
@yashwin, I've made some changes to address your comments. Can you have another look?

File: client/a8c-for-agencies/sections/marketplace/lib/hosting.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+export function getHostingLogo( slug: string ) {
+	switch ( slug ) {
+		case 'pressable-hosting':
+			return <img src={ PressableLogo } alt="" />;
```
Comment (jkguidaven) [inline client/a8c-for-agencies/sections/marketplace/lib/hosting.tsx:48]: Added alt="" so the screen reader ignores this as this is a decorative image.

**Review by yashwin — APPROVED** [review]
Thanks for addressing my comments. Looks great now :shipit:

**Pull request merged**